packages/compiler-cli/src/ngtsc/typecheck/src/type_check_block.ts

```diff
@@ -456,6 +451,31 @@ class TcbDirectiveInputsOp extends TcbOp {
           // declared in a `@Directive` or `@Component` decorator's `inputs` property) there is no
           // assignment target available, so this field is skipped.
           continue;
+        } else if (this.dir.restrictedInputFields.has(fieldName)) {
+          if (this.tcb.env.config.checkAccessModifiersForInputBindings) {
+            // To get errors assign directly to the fields on the instance, using property access
+            // when possible. String literal fields may not be valid JS identifiers so we use
+            // literal element access instead for those cases.
+            target = this.dir.stringLiteralInputFields.has(fieldName) ?
+                ts.createElementAccess(dirId, ts.createStringLiteral(fieldName)) :
+                ts.createPropertyAccess(dirId, ts.createIdentifier(fieldName));
+          } else {
+            // To ignore errors, assign to temp variable with type of the field
+            const id = this.tcb.allocateId();
+            const dirTypeRef = this.tcb.env.referenceType(this.dir.ref);
+            if (!ts.isTypeReferenceNode(dirTypeRef)) {
+              throw new Error(
+                  `Expected TypeReferenceNode from reference to ${this.dir.ref.debugName}`);
+            }
+            const type = ts.createIndexedAccessTypeNode(
+                ts.createTypeQueryNode(dirId as ts.Identifier),
+                ts.createLiteralTypeNode(ts.createStringLiteral(fieldName)));
+            const temp = tsCreateVariable(id, ts.createNonNullExpression(ts.createNull()), type);
```

Comment:
This can probably be:

```suggestion
            const temp = tsDeclareVariable(id, type);
```